build.sbt

```diff
@@ -13,14 +16,25 @@ lazy val slickMacros = (project in file("slick-macros"))
     libraryDependencies ++= Seq(
       "com.typesafe.slick" %% "slick" % "3.3.0",
       "com.chuusai" %% "shapeless" % "2.3.3",
-      scalaOrganization.value % "scala-reflect" % scalaVersion.value % Provided,
+      scalaOrganization.value % "scala-reflect" % scalaVersion.value % Provided
     ),
-    addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full),
-    scalacOptions --= Seq("-Ywarn-unused:patvars", "-Wunused:patvars"),
+    addCompilerPlugin(
+      "org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full
+    ),
+    // `val q"... $_ ..." = tree` fires "unused pattern variables" warning
```

Comment:
Well, I don't follow this line or the comment at all...

Reply:
You can use pattern bindings in `val` declarations, like `val List(one, two) = List(1, 2)`

I'm using [quasiquotes](https://docs.scala-lang.org/overviews/quasiquotes/intro.html) to parse the existing source AST and generate more AST. Those support pattern bindings, like

```scala
val q"class $className {}" = q"class FooBar {}"
assert(className == tq"FooBar") // tq = the text is parsed as a `Type` rather than a `Term`
```

When I discard parts of the AST that aren't useful to me by binding them with `$_`, the compiler lint "Unused pattern variable" is triggered, which cases compile to fail since I'm using `-Xfatal-warnings`. If I don't bind them, the binding throws a `MatchException`, as if I had done `val List(one) = List(1, 2)`